test/library/dumplogs.go

```diff
@@ -6,12 +6,15 @@ package library
 import (
 	"bufio"
 	"context"
+	"fmt"
 	"testing"
 	"time"
 
+	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
 )
 
 // DumpLogs is meant to be called in a `defer` to dump the logs of components in the cluster on a test failure.
```

Comment:
Should we make `DumpLogs` part of the `t.Cleanup` in `IntegrationEnv` instead of just when pods restart?

Reply:
I think I'm going to add a layer of log archiving at the CI level and then get rid of DumpLogs altogether. The hard part about what it's doing now is that it's trying not to clog stdout with a million log lines when you have a bunch of failed tests.

In _most_ cases when a test fails, it's not useful to dump all the logs from every Pinniped pod in the cluster. There are a few specific tests currently that opt-in to that behavior because they don't otherwise have good diagnostic output (e.g., the "full integration" E2E test).

I think it will be better if we make this problem going away by making sure we always have all the logs for post-hoc debugging.